oracles/src/domain/llm/anthropic_llm.py

```diff
@@ -0,0 +1,132 @@
+import json
+import time
+import backoff
+from typing import List
+from typing import Union
+from typing import Optional
+
+
+from openai.types.completion_usage import CompletionUsage
+from openai.types.chat import ChatCompletion
+from openai.types.chat import ChatCompletionToolParam
+from openai.types.chat.chat_completion import Choice
+from openai.types.chat.chat_completion import ChatCompletion
+from openai.types.chat.chat_completion_message import ChatCompletionMessage
+from openai.types.chat.chat_completion_message_tool_call import Function
+from openai.types.chat.chat_completion_message_tool_call import (
+    ChatCompletionMessageToolCall,
+)
+import anthropic
+from anthropic import NotGiven
+from anthropic import NOT_GIVEN
+from anthropic import AsyncAnthropic
+from anthropic.types import Message
+
+from src.entities import Chat
+
+import settings
+
+
+@backoff.on_exception(
+    backoff.expo, (anthropic.RateLimitError, anthropic.APITimeoutError), max_tries=3
+)
+async def execute(chat: Chat) -> Optional[ChatCompletion]:
+    client = AsyncAnthropic(api_key=settings.ANTHROPIC_API_KEY)
+    system_prompt = NOT_GIVEN
+    messages = chat.messages
+
+    if messages[0]["role"] == "system":
```

Comment:
maybe im paranoid but do we need to check the messages length here as well just in case?